sterror/sterror.go

```diff
@@ -0,0 +1,157 @@
+package sterror
+
+import (
+	"errors"
+)
+
+type Op string
+
+func (o Op) String() string {
+	return string(o)
+}
+
+type Scope string
+
+func (s Scope) String() string {
+	return string(s)
+}
+
+type Info string
+
+func (i Info) String() string {
+	return string(i)
+}
+
+const (
+	colon     = ": "
+	hyphen    = " - "
+	separator = "\n"
+)
+
+const (
+	Undefined Scope = "Undefined"
```

Comment:
As long as `Scope` is a string (not an enum) I think we do not need "Undefined" as this will be represented by "" (the empty string)

Reply:
Error() String Output would benefit from "Undefined" instead of "" as the default NewError() would always at least return Undefined instead of just an empty stringline in the Error Stack

Reply:
Good point, but this can then be handled in the Error() function itselfe, like `if e.Scope == "" {print something meaningful}`